### ISSUE TITLE
Unidentified issue with icinga2_web2_db database setup

### ISSUE THREAD

**Issue by MAccadia** [Icinga/icinga2-ansible#72]
During Icinga2 Web2 UI graphical setup on Ubuntu ; after `Database Resource` settings.

## Current Behavior

> Database Setup
> 
>  It seems that either the database you defined earlier does not yet exist and cannot be created using the provided access credentials, the database does not have the required schema to be operated by Icinga Web 2 or the provided access credentials do not have the sufficient permissions to access the database. Please provide appropriate access credentials to solve this.

## Expected Behavior

> Authentication Backend
>
>    As you've chosen to use a database for authentication all you need to do now is defining a name for your first authentication backend.

## Possible Solution

Granting all privileges to icinga2_web2_db_user as [suggested in documentation](https://www.icinga.com/docs/icingaweb2/latest/doc/02-Installation/#preparing-web-setup-on-debian) solves the problem. However, it might be good to identify the root cause.

```yaml
- name: Create Icinga2 Web Database User and configure Grants
  mysql_user:
    name: "{{ icinga2_web2_db_user }}"
    password: "{{ icinga2_web2_db_pass }}"
    state: present
    priv: "{{ icinga2_web2_db }}.*:ALL"
```

## Your Environment
```
ansible 2.4.3.0
python version = 2.7.12
Ubuntu 16.04.4 LTS
```

**Comment by aflatto** [comment]
Hi @MAccadia 
Can you share the error you got ? 
A screen shot would be very helpful and maybe even the ansible execution output of the relevant step.
Thank you

**Comment by MAccadia** [comment]
Hello, I wasn't able to reproduce the error. I guess I was using a wrong password / db name somehow. Sorry...

**Comment by MAccadia** [comment]
I run into the same issue today. Fresh install on Ubuntu VM.

I use :

```yaml
    - role: icinga2-ansible-web2-ui
      icinga2_db: "icinga2"
      icinga2_db_user: "icinga2"
      icinga2_db_pass: "**********"
      icinga2_web2_db: "icingaweb"
      icinga2_web2_db_user: "icingaweb"
      icinga2_web2_db_pass: "***********"
```

The playbook run without errors. When I got to http://IP//icingaweb2/setup I get this first :
![screenshot-2018-3-15 icinga web 1](https://user-images.githubusercontent.com/14163894/37466234-8ab1daf6-285d-11e8-9f51-c45acdc188e0.png)

and then

![screenshot-2018-3-15 icinga web](https://user-images.githubusercontent.com/14163894/37466270-9de0de74-285d-11e8-856e-2040927980fa.png)

On my server, I have :
```
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| icinga2            |
| icingaweb          |
| mysql              |
| performance_schema |
| sys                |
+--------------------+
6 rows in set (0,00 sec)

mysql> select user,host from mysql.user;
+------------------+---------------------+
| user             | host                |
+------------------+---------------------+
| repl             | %                   |
| root             | 127.0.0.1           |
| root             | ::1                 |
| debian-sys-maint | localhost           |
| icinga2          | localhost           |
| icingaweb        | localhost           |
| mysql.session    | localhost           |
| mysql.sys        | localhost           |
| root             | localhost           |
| root             | phi-monitoring-test |
+------------------+---------------------+
10 rows in set (0,00 sec)

mysql> show grants for 'icingaweb'@'localhost';
+---------------------------------------------------------------------------------------------------------------------------------------+
| Grants for icingaweb@localhost                                                                                                        |
+---------------------------------------------------------------------------------------------------------------------------------------+
| GRANT USAGE ON *.* TO 'icingaweb'@'localhost'                                                                                         |
| GRANT SELECT, INSERT, UPDATE, DELETE, DROP, INDEX, EXECUTE, CREATE VIEW ON `icingaweb`.* TO 'icingaweb'@'localhost' WITH GRANT OPTION |
+---------------------------------------------------------------------------------------------------------------------------------------+
2 rows in set (0,00 sec)
```

Then, If I run `GRANT ALL ON 'icingaweb'.* TO 'icingaweb'@'localhost'` and click `Back` and `Next` in IcingaWeb2 Setup. I get the expected :

![screenshot-2018-3-15 icinga web 2](https://user-images.githubusercontent.com/14163894/37466624-82c61bd0-285e-11e8-95b0-d26a7922dc9e.png)

**Comment by aflatto** [comment]
Hello @MAccadia 
The issue is fixed in #81 

Thank you for your input

**Comment by coom** [comment]
Same here ..

Created database like this :

```mysql
DROP DATABASE icinga_auth;
CREATE DATABASE icinga_auth;
GRANT ALL ON icinga_auth.* TO 'icinga_auth'@'localhost' IDENTIFIED BY '0zhzaoOmMs4PURZRYZnl';
FLUSH PRIVILEGES;
```

Then I click on "Validate configuration", the wizzard says that everything is OK, then I click next and :

It seems that either the database you defined earlier does not yet exist and cannot be created using the provided access credentials, the database does not have the required schema to be operated by Icinga Web 2 or the provided access credentials do not have the sufficient permissions to access the database. Please provide appropriate access credentials to solve this.

Do you have an idea?

**Comment by devchristianschroeder** [comment]
Same on my side

**Comment by MAccadia** [comment]
This issue has been fixed for me with #81. The only advice I can give is to double check your login/passwords.

**Comment by devchristianschroeder** [comment]
I imported the mysql schema manually into the database 'icinga2_web', assign ALL privileges to the database and user 'icinga' but the error existed. Then I changed the user name and password to root in the web configuration wizard and it worked. The root user has super privileges may be this is the point?

![capture2](https://user-images.githubusercontent.com/25476817/47082245-42af6980-d20d-11e8-8d23-e894d0433162.PNG)

![capture3](https://user-images.githubusercontent.com/25476817/47082253-4642f080-d20d-11e8-9a5f-bc5522e86526.PNG)

![capture](https://user-images.githubusercontent.com/25476817/47082112-ea786780-d20c-11e8-9a6e-6e6fd414570d.PNG)
